casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -145,10 +146,13 @@ object Genesis {
       bonds       <- getBonds[F](bondsFile, numValidators, genesisPath)
       timestamp   <- deployTimestamp.fold(Time[F].currentMillis)(_.pure[F])
       initial     = withoutContracts(bonds = bonds, timestamp = 1L, version = 0L, shardId = shardId)
+      validators  = bonds.map(bond => ProofOfStakeValidator(bond._1, bond._2)).toSeq
+      faucetCode  = if (faucet) Faucet.basicWalletFaucet(_) else Faucet.noopFaucet
```

Comment:
I feel like this is the critical time in which http://degoes.net/articles/destroy-all-ifs applies. We need to somehow make it such that we can't launch mainnet until this is toggled correctly.

Reply:
But the Boolean is not the core API, the `faucetCode` function is. The Boolean is pushed all the way to the edge of the program where we are reading things from the config file. The config necessarily has to be a simplification of the full API because we don't expect people to write code in the config file.
Also, in the config the default is `false` so we should just not have to specify anything to get the right behaviour for main net.